tools/common/Assembly.cs

```diff
@@ -313,7 +313,7 @@ void ProcessNativeReferenceOptions (NativeReferenceMetadata metadata)
 			}
 
 			// Don't add -force_load if the binding's SmartLink value is set and the static registrar is being used.
-			if (metadata.ForceLoad && !(metadata.SmartLink && App.Registrar == RegistrarMode.Static))
+			if (metadata.ForceLoad && !(metadata.SmartLink && (App.Registrar == RegistrarMode.Static || App.Registrar == RegistrarMode.ManagedStatic)))
```

Comment:
That could be a bit confusing, since there's also a `PartialStatic` entry in that enum, which shouldn't be included in this particular case (but it might sound like something that might or might not be in the extension method).